src/effects/backends/effectsbackendmanager.cpp

```diff
@@ -0,0 +1,115 @@
+#include "effects/backends/effectsbackendmanager.h"
+
+#include "control/controlobject.h"
+#include "effects/backends/builtin/builtinbackend.h"
+#include "effects/backends/effectprocessor.h"
+#ifdef __LILV__
+#include "effects/backends/lv2/lv2backend.h"
+#endif
+#include "effects/presets/effectpreset.h"
+
+EffectsBackendManager::EffectsBackendManager() {
+    m_pNumEffectsAvailable = std::make_unique<ControlObject>(
+            ConfigKey("[Master]", "num_effectsavailable"));
+    m_pNumEffectsAvailable->setReadOnly();
+
+    addBackend(EffectsBackendPointer(new BuiltInBackend()));
+#ifdef __LILV__
+    addBackend(EffectsBackendPointer(new LV2Backend()));
+#endif
+}
+
+void EffectsBackendManager::addBackend(EffectsBackendPointer pBackend) {
+    VERIFY_OR_DEBUG_ASSERT(pBackend) {
+        return;
+    }
+
+    m_effectsBackends.insert(pBackend->getType(), pBackend);
+
+    QList<QString> backendEffects = pBackend->getEffectIds();
+    for (const QString& effectId : backendEffects) {
+        m_manifests.append(pBackend->getManifest(effectId));
+    }
+
+    m_pNumEffectsAvailable->forceSet(m_manifests.size());
+
+    std::sort(m_manifests.begin(),
+            m_manifests.end(),
+            EffectManifest::alphabetize);
+}
+
+const QList<EffectManifestPointer> EffectsBackendManager::getManifestsForBackend(
+        EffectBackendType backendType) const {
+    auto pBackend = m_effectsBackends.value(backendType);
+    VERIFY_OR_DEBUG_ASSERT(pBackend) {
+        return QList<EffectManifestPointer>();
+    }
+    auto list = pBackend->getManifests();
+    std::sort(list.begin(), list.end(), EffectManifest::alphabetize);
+    return list;
+}
+
+EffectManifestPointer EffectsBackendManager::getManifestFromUniqueId(
+        const QString& uid) const {
+    if (kEffectDebugOutput) {
+        //qDebug() << "EffectsBackendManager::getManifestFromUniqueId" << uid;
+    }
+    if (uid.isEmpty()) {
+        // Do not DEBUG_ASSERT, this may be a valid request for a nullptr to
```

Comment:
it's generally bad practice / code smell to overload nullptr to mean things like "not present" or "unload."  Instead it's better to use std::optional

Reply:
Good point. This branch was started before we switched to requiring C++17. std::optional would be more appropriate now.

Reply:
Eh, I'm not so sure about this. If a pointer is already being used, it makes sense to use nullptr to mean "unload". With std::optional, this code would have to check for both nullopt and nullptr.